### ISSUE TITLE
removing common dbsnp with snpeff -> example 5 

### ISSUE THREAD

**Issue by mortunco** [pcingola/SnpEff#141]
Hi,

I would like to remove common dbsnp from my vcfs and thank you for setting up an tutorial for this case. However, during the removal of the dbsnp from the vcfs. I encountered with an error. My data is originated from ICGC vcf files for prostate cancer and the  interesting thing is their ID column is not empty so when I annotate my dbsnp, snpeff adds ";rs[somenumber]'. However, the last step of the dbsnp filtering is based on "! exists ID". but in my case, ID column is always filled up with an id which comes from raw untouched vcfs. 

What kind of filtration I should I in order to remove in my situation ? 

rare snp
```
3	110585271	c9a35b5e-2464-11e5-a4df-7335283a0828	G	T	.	SR;VUM	DP=74;MP=9.6e-01;GP=6.2e-10;TG=GG/GT;TP=9.6e-01;SG=GG/GG;SP=4.4e-02;SNP;ANN=T|intergenic_region|MODIFIER|Y_RNA-RP11-553A10.1|ENSG00000201426-ENSG00000206532|intergenic_region|ENSG00000201426-ENSG00000206532|||n.110585271G>T||||||	GT:FAZ:FCZ:FGZ:FTZ:RAZ:RCZ:RGZ:RTZ:PM	0|0:0:0:13:0:0:0:18:0:0.0e+00	0|1:0:0:16:1:0:0:23:3:9.3e-02
```

Found common dbsnp.
```
4	69536844	c9c18a3e-2464-11e5-a4df-7335283a0828;rs1120265	C	T	.	MN;MNP;SE;VUM	DP=58;MP=9.9e-01;GP=1.2e-02;TG=CC/CT;TP=9.9e-01;SG=CT/CC;SP=1.2e-02;SNP;ASP;CAF=0.5557,0.4443;COMMON=1;GENEINFO=UGT2B15:7366;KGPhase3;PM;PMC;R5;RS=1120265;RSPOS=69536844;RV;SAO=0;SLO;SSR=0;VC=SNV;VP=0x050128020005100026000100;WGT=1;dbSNPBuildID=86;ANN=T|upstream_gene_variant|MODIFIER|UGT2B15|ENSG00000196620|transcript|ENST00000338206|protein_coding||c.-508G>A|||||498|,T|intergenic_region|MODIFIER|UGT2B15-RP11-1267H10.4|ENSG00000196620-ENSG00000248635|intergenic_region|ENSG00000196620-ENSG00000248635|||n.69536844C>T||||||	GT:FAZ:FCZ:FGZ:FTZ:RAZ:RCZ:RGZ:RTZ:PM	0|0:0:9:0:0:0:8:0:1:5.6e-02	0|1:0:16:0:0:0:18:0:6:1.5e-01
```

Thank you for your help.

Best,

Tunc.

**Comment by mortunco** [comment]
Hi,

Let me rephrase my self.

I checked snpEff documentation again.  Could you guide me to make it exits ID and not rs.

ID not ~= 'rs

```
Has an ID and it matches the regulat expression 'rs':
 "(exists ID) & ( ID =~ 'rs' )" 

```

**Comment by mortunco** [comment]
YES YES YES.

cat snpeff_out.vcf | java -jar $snpSiftjar filter "(exists ID) & ( ID != 'rs')"

Could you verify me if this is right?

Sorry for my over excitement.

T.

**Comment by pcingola** [comment]
The examples are for educational purposes and would have to be customized for your specific purposes.
I think you have several options:

i) You can clean up the FILTER field previous to annotating with dbSnp. Obviously if 
you need to keep the ICGC ids, this would not be an option.

ii) Alternatively, you can also use a regex match (e.g. '=~'). You can attempt to do something 
like "rs" as long as you know that it does NOT appear in you ICGC Ids, but it looks like "rs" 
might appear there. A better option would be to match against "^rs" (i.e. starting with 'rs') or ";rs" (spearator plus 'rs):

    (exists ID) & !( ID =~ '^rs' ) & !( ID =~ ';rs' )

Remember that you can negate the expression using "!".
I hope this helps

**Comment by mortunco** [comment]
Dear Pablo,

This was the exact answer that I needed. 

Thank you for your fast respond and help.

Best,

Tunc.
